### PR TITLE
audit when the MFA for admin actions requirement is changed

### PR DESCRIPTION
**Pull request by capnspacehook**
Per the docs MFA is only required for admin actions when the cluster second factor is set to webauthn only: https://goteleport.com/docs/access-controls/guides/mfa-for-admin-actions/#require-mfa-for-administrative-actions

Fixes https://github.com/gravitational/teleport-private/issues/1461.

changelog: Add audit event field describing if the "MFA for admin actions" requirement changed

### CODE REVIEW

**Review by hugoShaka — APPROVED** [review]

File: api/proto/teleport/legacy/types/events/events.proto [hunk 1]
```diff
@@ -6162,6 +6162,16 @@ message AuthPreferenceUpdate {
     (gogoproto.embed) = true,
     (gogoproto.jsontag) = ""
   ];
+
+  // AdminActionsMFAChanged indicates whether this update changed if
+  // MFA is required for admin actions or not
+  AdminActionsMFAStatus AdminActionsMFAChanged = 5 [(gogoproto.jsontag) = "admin_actions_mfa_changed"];
+}
+
+enum AdminActionsMFAStatus {
```
[the remark this answers is not in the document]
Comment (capnspacehook) [reply api/proto/teleport/legacy/types/events/events.proto:6171]: The zero value being unspecified I could go either way on. I get why it's best practice but having it default to 'no changes' here is useful and makes sense. I'm fine with making the change though.

Prefixing enum values I can do as well, but it creates very ugly symbol names. The protobuf code generator already adds prefixes to enum names, ie `AdminActionsMFAStatus_ENABLED`. After manually adding a prefix, the Go symbol name becomes `AdminActionsMFAStatus_ADMIN_ACTIONS_MFA_STATUS_ENABLED` which seems repetitive and verbose with no clear upside IMO.

[the remark this answers is not in the document]
Comment (capnspacehook) [reply api/proto/teleport/legacy/types/events/events.proto:6171]: I added an 'invalid' zero value, so you want it renamed to 'unspecified'?

[the remark this answers is not in the document]
Comment (capnspacehook) [reply api/proto/teleport/legacy/types/events/events.proto:6171]: Done

File: lib/auth/clusterconfig/clusterconfigv1/service.go [hunk 2]
```diff
@@ -318,18 +357,29 @@ func (s *Service) ResetAuthPreference(ctx context.Context, _ *clusterconfigpb.Re
...
 
-		if err := s.emitter.EmitAuditEvent(ctx, &apievents.AuthPreferenceUpdate{
+		adminActionsStatus := apievents.AdminActionsMFAStatus_ADMIN_ACTIONS_MFA_STATUS_UNCHANGED
+		if oldSecondFactor != newSecondFactor {
+			if newSecondFactor == constants.SecondFactorWebauthn {
+				adminActionsStatus = apievents.AdminActionsMFAStatus_ADMIN_ACTIONS_MFA_STATUS_ENABLED
+			} else if oldSecondFactor == constants.SecondFactorWebauthn {
+				adminActionsStatus = apievents.AdminActionsMFAStatus_ADMIN_ACTIONS_MFA_STATUS_DISABLED
+			}
+		}
```
[the remark this answers is not in the document]
Comment (capnspacehook) [reply lib/auth/clusterconfig/clusterconfigv1/service.go]: Done

File: api/proto/teleport/legacy/types/events/events.proto [hunk 3]
```diff
@@ -6162,6 +6162,17 @@ message AuthPreferenceUpdate {
     (gogoproto.embed) = true,
     (gogoproto.jsontag) = ""
   ];
+
+  // AdminActionsMFAChanged indicates whether this update changed if
+  // MFA is required for admin actions or not
```
[the remark this answers is not in the document]
Comment (capnspacehook) [reply api/proto/teleport/legacy/types/events/events.proto]: Sounds good, done

File: api/proto/teleport/legacy/types/events/events.proto [hunk 4]
```diff
@@ -6162,6 +6162,17 @@ message AuthPreferenceUpdate {
...
+  AdminActionsMFAStatus AdminActionsMFAChanged = 5 [(gogoproto.jsontag) = "admin_actions_mfa_changed"];
```
[the remark this answers is not in the document]
Comment (capnspacehook) [reply api/proto/teleport/legacy/types/events/events.proto]: Sounds fine, done

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**